packages/components/src/internal/PopupPortal.ts

```diff
@@ -0,0 +1,30 @@
+import { Children, PropsWithChildren, ReactNode } from 'react';
+import { createPortal } from 'react-dom';
+
+export const popupRootId = 'modal-root';
+
+/**
+ * Popups, modals, etc. are rendered into the popup root element.
+ * This function returns that element, creating it if it does not exist.
+ */
+export const getPopupRoot = (): HTMLElement => {
+  let popupRoot = document.getElementById(popupRootId);
+  if (!popupRoot) {
+    popupRoot = document.createElement('div');
+    popupRoot.id = popupRootId;
+    popupRoot.role = 'complementary';
+    document.body.append(popupRoot);
+  }
+  return popupRoot;
+};
+
+type PopupPortalProps = PropsWithChildren<{}>;
```

Comment:
export type plz

Reply:
@nawatts ^

Reply:
Oops, missed this in https://github.com/DataBiosphere/terra-ui/pull/4234/commits/ee07688c47955f1b581f1a9d5798a3079bbddc0b.

Reply:
Exported in https://github.com/DataBiosphere/terra-ui/pull/4234/commits/8cd41bf1bb0ae06e3fcb53d46eab84a5041facf3.